packages/iov-socket/src/queueingstreamingsocket.ts

```diff
@@ -0,0 +1,32 @@
+// tslint:disable:readonly-array readonly-keyword
+import { StreamingSocket } from "./streamingsocket";
+
+/**
+ * A StreamingSocket that can queue requests.
+ */
+export class QueueingStreamingSocket extends StreamingSocket {
+  private queue: string[] = [];
+
+  public getQueueLength(): number {
+    return this.queue.length;
+  }
+
+  public async queueRequest(data: string): Promise<void> {
```

Comment:
why should `queueRequest` bother about the processing? What is processes here can very well be unrelated to the enqueued request. I think this should just trigger `this.processQueue` without waiting for the result.

We might need other ways of error reporting later on, but right now I don't see how processQueue could throw/reject

Reply:
I also had this thought, I'm just scared of leaving promises unhandled in something changes later on and we don't update this. Another option would be to leave out the processing altogether and rely on the caller to do both? (Then the caller can leave the promise unhandled if they really want to.)